### PR TITLE
James's Work 21st March

### PR DESCRIPTION
**Pull request by odscjames**
Bits of work, each in own commit:

  *  Statistic: % of type of interest statement in ownership-or-control statement
  *  Statistic: % of Persons of each type
  *  Updated standard Schema to V0.1
  *  Statistic: % of ownership-or-control statements with unspecified interestedParty
  *  Statistic: % of Entities of each type
  *  Additional Checks: Each statement must have an entirely unique identifier

### CODE REVIEW

**Comment by odscjames** [comment]
#7 needs fixing, then this needs rebasing and work

**Comment by odscjames** [comment]
Ok, ready for review - each commit is a separate bit of work, so review one commit at a time maybe?

File: libcovebods/lib/common_checks.py [hunk 1]
```diff
@@ -40,15 +86,18 @@ def __init__(self, json_data):
         self.person_statements_seen_in_ownership_or_control_statement = []
         self.entity_statements_seen = []
         self.entity_statements_seen_in_ownership_or_control_statement = []
+        self.ownership_or_control_statements_seen = []
```
Comment (Bjwebb) [inline libcovebods/lib/common_checks.py]: `self.ownership_or_control_statements_seen` and `self.statement_ids_counted` are declared in more than one place, is this deliberate?

File: libcovebods/lib/common_checks.py [hunk 2]
```diff
@@ -238,3 +257,10 @@ def _check_ownership_or_control_statement_second_pass(self, statement):
                         'entity_statement_missing': subject_described_by_entity_statement,
                         'seen_in_ownership_or_control_statement': statement.get('statementID'),
                     })
+
+    def _add_statement_ids_to_statement_ids_counted(self, statement_ids):
+        for statement_id in statement_ids:
+            if statement_id in self.statement_ids_counted:
+                self.statement_ids_counted[statement_id] += 1
+            else:
+                self.statement_ids_counted[statement_id] = 1
```
Comment (Bjwebb) [inline libcovebods/lib/common_checks.py]: `defaultdict` could make this shorter.

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,9 +7,18 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+
+- Statistic: % of Persons of each type
```
Comment (Bjwebb) [inline CHANGELOG.md]: The percentage is not actually calculated in lib-cove-bods, right?

**Comment by odscjames** [comment]
Thanks! :-)

File: libcovebods/lib/common_checks.py (hunk 1 above)
Comment (odscjames) [reply libcovebods/lib/common_checks.py]: Yeah, Pep complained if I didn't do it in init, and I wanted it in Run incase someone called run twice on the same object. Maybe my use of the "run" function is a bit odd here, happy to discuss.

File: libcovebods/lib/common_checks.py (hunk 2 above)
Comment (odscjames) [reply libcovebods/lib/common_checks.py]: Had not seen that before, nice!

File: CHANGELOG.md (hunk 3 above)
Comment (odscjames) [reply CHANGELOG.md]: No, a lot of the stat's are just raw numbers. But good point, the changelog is technically wrong then

**Pull request merged**